src/vellum/workflows/nodes/core/map_node/node.py

```diff
@@ -85,12 +101,27 @@ def run(self) -> Outputs:
                     )
         except Empty:
             pass
-
         return self.Outputs(**mapped_items)
 
+    def _context_run_subworkflow(
+        self, *, item: MapNodeItemType, index: int, parent_context: Optional[ParentContext] = None
+    ) -> None:
+        parent_context = parent_context or self._context.parent_context
+        with execution_context(parent_context=parent_context):
+            self._run_subworkflow(item=item, index=index)
+
     def _run_subworkflow(self, *, item: MapNodeItemType, index: int) -> None:
-        subworkflow = self.subworkflow(parent_state=self.state, context=self._context)
-        events = subworkflow.stream(inputs=self.SubworkflowInputs(index=index, item=item, all_items=self.items))
+        context = WorkflowContext(
+            _vellum_client=self._context._vellum_client, _parent_context=self._context.parent_context
+        )
+        if self._context._event_queue:
+            context._register_event_queue(self._context._event_queue)
```

Comment:
I wouldn't expect to need this, since `self.subworkflow` will have its own event queueing.